### ISSUE TITLE
[Documentation] Alternate installation method using virtual enviorments for python and node.

### ISSUE THREAD

**Issue by v3ss0n** [RocketMap/RocketMap#1042]
I am not sure where to put this so i made this as an issue , not sure if it fits in Basic or Advance linux instructions.
## Motivation

Setup instructions in docs are too hard to follow for linux beginners , and different linux will have different problems , since this requires latest python 2.7 and not many distros have that.
This should work in many distros and could easily adapt it to any other linux distros  since this method do not require building python by users themselves and they just copy and paste after changing `user` 
 `password` and `apikey` .
So i made this script (too bored to make it accept input  for user /pass/apikey , can easily add it)

``` bash
sudo apt-get -y update && sudo apt-get -y install build-essential git
wget -qO- https://raw.githubusercontent.com/creationix/nvm/v0.31.6/install.sh | bash
export NVM_DIR="$HOME/.nvm"
[ -s "$NVM_DIR/nvm.sh" ] && . "$NVM_DIR/nvm.sh" # This loads nvm
nvm install node
wget https://repo.continuum.io/miniconda/Miniconda2-latest-Linux-x86_64.sh -O ~/miniconda.sh
bash ~/miniconda.sh -b -p $HOME/miniconda
export PATH="$HOME/miniconda/bin:$PATH"
source ~/miniconda/bin/activate root
git clone -b develop https://github.com/PokemonGoMap/PokemonGo-Map.git PoGoMap
cd  PoGoMap
pip install -r requirements.txt
source ~/.bashrc
npm install && npm run build
source ~/miniconda/bin/activate root
python ./runserver.py -a ptc -u "Username" -p "Password" \
 -l "Location(town , city , street)" -st 3 -k "Google API Key"
```

now every ubuntu users can just easily copy and paste it. For all other linux version , they just need to switch the first line , adapt it for their package managers .

**Comment by pyr0ball** [comment]
This would work fine on newer distributions, but not on VPS installations that use Debian 8 or older, as the python / pip / node versions available via packages are too far out of date. They must be built and installed from source, which is (unfortunately) not easy. We could probably build a unified script that would use different install parameters based on the version of linux it detects, but we'll need a lot of work and feedback on that. also lots of "testing in the wild"

**Comment by v3ss0n** [comment]
> Debian 8 or older, as the python / pip / node versions available via packages are too far out of date. They must be built and installed from source, which is (unfortunately) not easy. We could probably build a unified script that would use different install parameters based on the version of linux it detects, but we'll need a lot of work and feedback on that. also lots of "testing in the wild"

no , this setup install  python and pip  from conda and node from nvm , which is self contained version of their own runtimes in virtual enviorment . This setup gives latest for both python and node. Please test first. This is actually designed to install latests on old OS.

**Comment by m3g4mInD** [comment]
I have tested and work 99% thanks for sharing to us.

**Comment by v3ss0n** [comment]
let me know the 1% :)

**Comment by pyr0ball** [comment]
Will test and report, ty!

**Comment by v3ss0n** [comment]
should i add PR with doc for this ?

**Comment by sebastienvercammen** [comment]
Replace `nvm/v0.31.6` to automatically get the latest version of nvm, and convert this snippet into an actual (readable, properly formatted) bash install script, and we can add to the docs.

In its current state, it'll eventually cause problems when things get out of date, or when people start copy/pasting but they don't have `apt-get` on their distro.

I'm also not sure about Miniconda, needs a proper discussion.

**Comment by sebastienvercammen** [comment]
Closed, this is a very old issue. Any feature requests are now handled via http://vote.devkat.org and if any issue reported in the issue is still related to current code, a new GitHub issue should be opened.
